File: decorator/en_us/python/src/circle.py
from geometric import Geometric
from math import *

""" Basic implementation of Geometric """
class Circle(Geometric):
    """
    Needs a radius to be instantiated,
    because all operations must use this property
    """
    def __init__(self, radius):
        self.radius = radius

    def calculate_area(self):
        self.area = (pi * self.radius ** 2)
        print("Area = " + str(self.area))

    def calculate_perimeter(self):
        self.perimeter = (2 * pi * self.radius)
        print("Perimeter = " + str(self.perimeter))

File: decorator/en_us/java/Circle.java
package java;

/**
 * Basic implementation of Geometric
 */
public class Circle extends Geometric {

    private float radius;

    /**
     * Needs a radius to be instantiated,
     * because all operations must use this property
     */
    public Circle(float radius) {
        this.setRadius(radius);
    }

    private void setRadius(float radius) {
        this.radius = radius;
    }

    private double getRadius() {
        return radius;
    }

    public void calculateArea() {
        double area = Math.PI * this.getRadius() * 2;
        System.out.println("Area = " + area);
    }

    public void calculatePerimeter() {
        double perimeter = 2 * Math.PI * this.getRadius();
        System.out.println("Perimeter = " + perimeter);
    }

}
